.github/workflows/release.yaml

```diff
@@ -1,205 +1,207 @@
-                    token: ${{ secrets.APIFY_SERVICE_ACCOUNT_GITHUB_TOKEN }}
-
-            -   name: Use Node.js 22
-                uses: actions/setup-node@v4
-                with:
-                    node-version: 22
-
-            -   name: Enable corepack
-                run: |
-                    corepack enable
-                    corepack prepare yarn@stable --activate
-
-            -   name: Activate cache for Node.js 22
-                uses: actions/setup-node@v4
-                with:
-                    cache: yarn
-
-            -   name: Update package version in package.json
-                run: npm version --no-git-tag-version --allow-same-version ${{ needs.release_metadata.outputs.version_number }}
-
-            -   name: Update CHANGELOG.md
-                uses: DamianReeves/write-file-action@master
-                with:
-                    path: CHANGELOG.md
-                    write-mode: overwrite
-                    contents: ${{ needs.release_metadata.outputs.changelog }}
-
-            -   name: Commit changes
-                id: commit
-                uses: EndBug/add-and-commit@v9
-                with:
-                    author_name: Apify Release Bot
-                    author_email: noreply@apify.com
-                    message: "chore(release): Update changelog and package version [skip ci]"
-
-    create_github_release:
-        name: Create github release
-        needs: [release_metadata, update_changelog]
-        runs-on: ubuntu-latest
+        id: release_metadata
+        with:
+          release_type: ${{ inputs.release_type }}
+          custom_version: ${{ inputs.custom_version }}
+          existing_changelog_path: CHANGELOG.md
+
+  wait_for_checks:
+    name: Wait for code checks to pass
+    runs-on: ubuntu-latest
+    steps:
+      - uses: lewagon/wait-on-check-action@v1.3.4
+        with:
+          ref: ${{ github.ref }}
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          check-regexp: (Build & Test .*|Test Python template support|Lint|Docs build)
+          wait-interval: 5
+
+  update_changelog:
+    needs: [release_metadata, wait_for_checks]
+    name: Update changelog
+    runs-on: ubuntu-latest
+    outputs:
+      changelog_commitish: ${{ steps.commit.outputs.commit_long_sha || github.sha }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.APIFY_SERVICE_ACCOUNT_GITHUB_TOKEN }}
+
+      - name: Use Node.js 22
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22
+
+      - name: Enable corepack
+        run: |
+          corepack enable
+          corepack prepare yarn@stable --activate
+
+      - name: Activate cache for Node.js 22
+        uses: actions/setup-node@v4
+        with:
+          cache: yarn
+
+      - name: Update package version in package.json
+        run: yarn version ${{ needs.release_metadata.outputs.version_number }}
```

Comment:
Is this really equivalent to `--no-git-tag-version --allow-same-version` for `npm version`?